tests/e2e/specs/miscellaneous/CreateWorkspaceWithExistedNameFactory.spec.ts

```diff
@@ -0,0 +1,149 @@
+		await browserTabsUtil.closeAllTabsExceptCurrent();
+
+		await dashboard.clickCreateWorkspaceButton();
+		await createWorkspace.waitPage();
+		expect(await createWorkspace.isCreateNewWorkspaceCheckboxChecked()).to.be.true;
+	}
+
+	async function waitWorkspaceReadiness(): Promise<void> {
+		const originalWindowHandle: string = await browserTabsUtil.getCurrentWindowHandle();
+		await browserTabsUtil.waitAndSwitchToAnotherWindow(originalWindowHandle, TIMEOUT_CONSTANTS.TS_IDE_LOAD_TIMEOUT);
+
+		await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+
+		registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
+
+		await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+		projectSection = await projectAndFileTests.getProjectViewSession();
+		expect(await projectAndFileTests.getProjectTreeItem(projectSection, projectName), 'Project folder was not imported').not.undefined;
+		expect(
+			await projectAndFileTests.getProjectTreeItem(projectSection, BASE_TEST_CONSTANTS.TS_SELENIUM_PROJECT_ROOT_FILE_NAME),
+			'Project files were not imported'
+		).not.undefined;
+	}
+
+	test('Verify "Create New" is off and a new workspace is created', async function (): Promise<void> {
+		await waitDashboardPage();
+
+		await createWorkspace.setGitRepositoryUrl(factoryUrl);
+		expect(await createWorkspace.isCreateNewWorkspaceCheckboxChecked()).to.be.false;
+		expect(await createWorkspace.getGitRepositoryUrl()).to.be.equal(factoryUrl);
+		await createWorkspace.clickOnCreateAndOpenButton();
+		await createWorkspace.performTrustAuthorPopup();
+
+		await waitWorkspaceReadiness();
+
+		firstWorkspaceName = WorkspaceHandlingTests.getWorkspaceName();
+	});
+
+	test('Verify existing workspace opens instead of creating a duplicate', async function (): Promise<void> {
+		await waitDashboardPage();
+
+		await createWorkspace.setGitRepositoryUrl(factoryUrl);
+		expect(await createWorkspace.getGitRepositoryUrl()).to.be.equal(factoryUrl);
+		await createWorkspace.clickOnCreateAndOpenButton();
+		await createWorkspace.performTrustAuthorPopup();
+
+		await waitWorkspaceReadiness();
+
+		expect(WorkspaceHandlingTests.getWorkspaceName()).to.be.equal(firstWorkspaceName);
+	});
+
+	test('Ensure `policies.create=perclick` is appended and a new workspace is created without warnings', async function (): Promise<void> {
+		await waitDashboardPage();
+
+		await createWorkspace.setGitRepositoryUrl(factoryUrl);
+		await createWorkspace.setCreateNewWorkspaceCheckbox(true);
+		expect(await createWorkspace.getGitRepositoryUrl()).to.be.equal(factoryUrl + '?policies.create=perclick');
+
+		await createWorkspace.clickOnCreateAndOpenButton();
+		await createWorkspace.performTrustAuthorPopup();
+
+		await waitWorkspaceReadiness();
+
+		secondWorkspaceName = WorkspaceHandlingTests.getWorkspaceName();
+		expect(secondWorkspaceName).to.have.string(firstWorkspaceName);
+		expect(secondWorkspaceName.length).to.be.equal(firstWorkspaceName.length + 5);
```

Comment:
IMHO, It would improve the test coverage if we could check for a workspace name like: `firstWorkspaceName + "-z7vn"` using a regexp.